slides/week0/week0.org

```diff
@@ -0,0 +1,397 @@
+#+BEGIN_SRC C++ -n
+  int main() {
+    // <==== HERE =====
+  }
+#+END_SRC
+
+* ;
+#+BEGIN_NOTES
+Run this code to show them that the compiler shows an error on a different line
+than it is actually on.
+#include <iostream>
+using namespace std;
+int main() {
+    cout <<"Hello "
+    cout << "World!";
+    return 0;
+}
+#+END_NOTES
+- A semicolon defines the end of a line of code
+- Think of it like a period in a sentence
+- Almost every line of code will end with a semicolon
+    - A missing semicolon results in an error message that can point to the wrong line
+#+BEGIN_SRC C++ -n
+  int main() {
+    doSomething;
+  }
+#+END_SRC
+
+* Looking back at Hello World!
+- our code is in {}
+- *cout* tells the computer to write the letters to the screen
+- *<<* tells the computer what to write out
+- line 1: inclues a header file (covered later)
+- line 4: defines the main function
+- line 6: tells the computer that the program ran successfully
+#+BEGIN_SRC C++ -n
+  #include <iostream>
+  using namespace std;
+
+  int main() {
+    cout << "Hello World!";
+    return 0;
+  }
+#+END_SRC
+
+* Most useful program ever
+#+BEGIN_NOTES
+This should naturally lead into why variables are useful. Without them
+all programs would be deterministic.
+#+END_NOTES
+#+BEGIN_SRC C++ -n
+  int main() {
+    cout << 2 + 2;
+    return 0;
+  }
+#+END_SRC
+
+* Variables
+- Alias for a value
+- the value can be changed without changing the alias
+
+* Primitive Variables Types
+| Name | Description |
+|-------+------|
+| =bool= | Can be either *true* or *false*|
+| =char= | A number from -128 to 127 or a character like 'c' or '+'|
+| =int= | Any whole number from -2,147,483,648	2,147,483,647|
+| =double= | any number from 2.2e-308 to 1.8e+308 with decimals|
+- What makes a variable primitive?
+
+* Primitive Variables?
+[[file:https://i.imgur.com/Wo0ovX5.jpg]]
+- But Actually... It means that it cannot be divided into smaller parts
+- A number is just a number
+
+* Important Symbols
+#+BEGIN_NOTES
+No lengthy explanation of << and stdout
+#+END_NOTES
+| symbol | meaning |
+|-------+------|
+| =//= | Signifies a comment, The computer will not try to compile that line |
+| === | assignment operator, value of right is assigned to variable on left |
+| =<<= | sends something to the OS which displays it in the terminal |
+
+* Math Operators
+- order of operations still applies
+- () > *,/ > +,-
+| symbol | meaning |
+|-------+------|
+| =()= | =groups a statement= |
+| =*= | =multiplies two numbers= |
+| =/= | =divides a number by another= |
+| =+= | =adds two numbers together= |
+| =-= | =subtracts two numbers= |
+
+* Variable Syntax
+| =type= | =name= | === | =value= | ; |
+|-------+------|
+| =bool= | =var= | = | true | ; |
+| =char= | =character= | = | 'c' | ; |
+| =int= | =intVar= | = | 2 | ; |
+| =double= | =double_var= | = | 2.2647 | ; |
+
+* Variables
+- =my_var= now represents the value 2
+#+BEGIN_SRC C++ -n
```

Comment:
Let me know if it doesn't work for any reason, or if you have any problems with it! Source code block execution is the killer feature of org (IMO) because you can do so much with it. It is pretty badly documented though, which is it's main problem...